tests/model/test_model.py

```diff
@@ -1978,25 +1978,72 @@ def test__update_topic_index(
             "vary_each_msg",
             "visual_notification_status",
             "types_when_notify_called",
+            "is_muted_stream",
+            "is_muted_topic",
         ],
         [
             (
                 5140,
                 {"flags": ["mentioned", "wildcard_mentioned"]},
                 True,
                 [],
+                None,
+                None,
             ),  # message_fixture sender_id is 5140
-            (5179, {"flags": ["mentioned"]}, False, ["stream", "private"]),
-            (5179, {"flags": ["wildcard_mentioned"]}, False, ["stream", "private"]),
-            (5179, {"flags": []}, True, ["stream", "private"]),
-            (5179, {"flags": []}, False, ["private"]),
+            (5179, {"flags": ["mentioned"]}, False, ["stream", "private"], None, None),
+            (
+                5179,
+                {"flags": ["wildcard_mentioned"]},
+                False,
+                ["stream", "private"],
+                None,
+                None,
+            ),
+            (5179, {"flags": []}, True, ["stream", "private"], None, None),
+            (5179, {"flags": []}, False, ["private"], None, None),
+            (
+                5179,
+                {"flags": ["has_alert_word"]},
+                False,
+                ["stream", "private"],
+                True,
+                True,
+            ),
+            (
+                5179,
+                {"flags": ["has_alert_word"]},
+                False,
+                ["stream", "private"],
+                True,
+                False,
+            ),
+            (
+                5179,
+                {"flags": ["has_alert_word"]},
+                False,
+                ["stream", "private"],
+                False,
+                True,
+            ),
+            (
+                5179,
+                {"flags": ["has_alert_word"]},
+                False,
+                ["stream", "private"],
+                False,
+                False,
+            ),
         ],
         ids=[
             "not_notified_since_self_message",
             "notified_stream_and_private_since_directly_mentioned",
             "notified_stream_and_private_since_wildcard_mentioned",
             "notified_stream_since_stream_has_desktop_notifications",
             "notified_private_since_private_message",
+            "not_notified_for_stream_since_topic/stream_both_muted",
+            "not_notified_for_stream_since_stream_muted",
+            "not_notified_for_stream_since_topic_muted",
+            "notified_for_stream_since_topic/stream_both_not_muted",
```

Comment:
This is a lot of ids to manually map to the cases above them. Given there are already 5 and we're going to have many more, it would be great to have an initial refactor commit which inlines the existing ones into the test cases.

Also see my comment on an earlier iteration of the PR regarding this.